File: Java_Codes/9. Multithreading/ExecutorFrameworkExample.java
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

public class ExecutorFrameworkExample {
    public static void executeExecutorFrameworkExample() {
        ExecutorService executor = Executors.newFixedThreadPool(3);

        Runnable task = () -> System.out.println("Task executed by: " + Thread.currentThread().getName());

        for (int i = 0; i < 5; i++) {
            executor.execute(task);
        }

        executor.shutdown();
    }

    public static void main(String[] args) {
        executeExecutorFrameworkExample();
    }
}

File: Python_Codes/9. Multithreading/ExecutorFrameworkExample.py
from concurrent.futures import ThreadPoolExecutor
import threading

def task():
    print(f"Task executed by: {threading.current_thread().name}")

def execute_executor_framework_example():
    # Create a thread pool with 3 threads
    with ThreadPoolExecutor(max_workers=3) as executor:
        for _ in range(5):
            executor.submit(task)

if __name__ == "__main__":
    execute_executor_framework_example()
